src/misc/rwqueue.cpp

```diff
@@ -65,16 +78,40 @@ bool RWQueue<T>::IsEmpty()
 }
 
 template <typename T>
-void RWQueue<T>::Enqueue(T&& item)
+bool RWQueue<T>::StoppedEnqueuing() const
+{
+	return is_stopped;
+}
+
+template <typename T>
+bool RWQueue<T>::StoppedDequeuing()
+{
+	if (is_stopped) {
+		// We're only done when there's nothing left
+		std::lock_guard<std::mutex> lock(mutex);
+		return queue.empty();
+	}
+	return false;
+}
```

Comment:
Isn't this API a little bit redundant? We already have `IsEmpty`, would only adding `IsStopped` suffice instead of `StoppedEnqueuing` and `StoppedDequeuing`? Especially `StoppedDequeuing` combined multiple checks into one call.

Reply:
Knocked out `IsDequeuing()` in favour of the letting the user combine calls. Now down to `Stop()` and `IsRunning()`.